staging/src/k8s.io/client-go/tools/record/events_cache.go

```diff
@@ -77,6 +77,8 @@ func getSpamKey(event *v1.Event) string {
 		event.InvolvedObject.Name,
 		string(event.InvolvedObject.UID),
 		event.InvolvedObject.APIVersion,
+		event.Type,
+		event.Reason,
```

Comment:
The spam filter is meant to reduce the spam caused by faulty objects (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/record/events_cache.go#L128-L129) on any of the events they produce. If we were to distinguish between type and reason, the rate limiting would be less effective than it is today.

Maybe it could be worth looking into making the spam filter aware of the event type so that there are dedicated priority seats available for critical events when there is a burst of both normal and critical events.

Reply:
Including only event type to the spam filter sounds good too. Update the PR accordingly.